src/reduced_basis/rb_parametrized.C

```diff
@@ -393,35 +395,45 @@ bool RBParametrized::check_if_valid_params(const RBParameters &params) const
+  std::string prev_param_name = "";
+  for (const auto & [param_name, sample_vec] : params)
     {
-      // Check every parameter value (including across samples) to ensure
-      // it's within the min/max range.
-      const bool outside_range =
-        ((param_val < get_parameter_min(param_name)) ||
-         (param_val > get_parameter_max(param_name)));
-      is_valid = is_valid && !outside_range;
-      if(outside_range && verbose_mode)
+      std::size_t sample_idx = 0;
+      const Real & min_value = get_parameter_min(param_name);
+      const Real & max_value = get_parameter_max(param_name);
+      for (const auto & value_vec : sample_vec)
         {
-          libMesh::out << "Warning: parameter " << param_name << " value="
-                       << param_val << " outside acceptable range: ("
-                       << get_parameter_min(param_name) << ", "
-                       << get_parameter_max(param_name)
-                       << ")";
-        }
-
-      // For discrete params, make sure params.get_value(param_name) is sufficiently
-      // close to one of the discrete parameter values.
-      if (is_discrete_parameter(param_name))
-        {
-          const bool is_value_discrete =
-            is_value_in_list(param_val,
-                             get_discrete_parameter_values().find(param_name)->second,
-                             TOLERANCE);
-          is_valid = is_valid && is_value_discrete;
-          if(!is_value_discrete && verbose_mode)
-            libMesh::out << "Warning: parameter " << param_name << " value="
-                         << param_val << " is not in discrete value list.";
+          for (const auto & value : value_vec)
+            {
+              // Check every parameter value (including across samples and vector-values)
+              // to ensure it's within the min/max range.
+              const bool outside_range = ((value < min_value) || (value > max_value));
+              is_valid = is_valid && !outside_range;
+              if(outside_range && verbose_mode)
+                {
+                  libMesh::out << "Warning: parameter " << param_name << " value="
+                               << value << " outside acceptable range: ("
+                               << min_value << ", " << max_value << ")";
+                }
+
+              // For discrete params, make sure params.get_value(param_name) is sufficiently
+              // close to one of the discrete parameter values.
+              // Note that vector-values not yet supported in discrete parameters,
+              // and the .get_sample_value() call will throw an error if the user
+              // tries to do it.
+              if (is_discrete_parameter(param_name))
+                {
+                  const bool is_value_discrete =
+                    is_value_in_list(params.get_sample_value(param_name, sample_idx),
+                                     get_discrete_parameter_values().find(param_name)->second,
```

Comment:
I see that this is just re-indentation, but (as a note to myself) it would be good to avoid this double-lookup idiom by doing:
```
if (auto it = get_discrete_parameter_values().find(param_name);
    it != get_discrete_parameter_values().end())
{
  const bool is_value_discrete =
    is_value_in_list(params.get_sample_value(param_name, sample_idx),
                     it->second,
                     ...
```

Reply:
Good eye, I've gone ahead and updated this.

In general I haven't really touched the discrete parameters.